### PR TITLE
Switch to library from static library…

### PR DESCRIPTION
**Pull request by russel**
…add back the version and soversion attributes.

This is, I believe the alternate solution to the warnings problem that @ximion was proposing. 

Overall, I think this is a better paradigm for D library builds.

### CODE REVIEW

**Comment by russel** [comment]
It seems the Travis-CI fails are to do with `-L-soname` being passed as an argument to the dmd and ldc2 commands. When I build here on Debian Sid, the ldmd2 command is used and it works fine with the `-L-soname` argument.

**Comment by russel** [comment]
Test fail due to macOS and -soname option. Cogitating.

**Comment by russel** [comment]
I think this problem has to be deemed a bug in Meson. I shall raise a bug there.

**Comment by ximion** [comment]
@russel Looks like this was fixed in https://github.com/mesonbuild/meson/commit/23e6200c1470c33426e3f99eba32ebed4e76ca3a

**Comment by russel** [comment]
I won't be able to check this till end of next week, sorry.

**Comment by russel** [comment]
@ximion Looks like that is right:
```
|> meson_head  --prefix=$HOME/Built ~/Repositories/Git/Forks/TinyEndian/
The Meson build system
Version: 0.48.0.dev1
Source dir: /home/users/russel/Repositories/Git/Forks/TinyEndian
Build dir: /Users/russel/BuildArea/TinyEndian
Build type: native build
Project name: tinyendian
Project version: 0.2.0
Native D compiler: ldc2 (llvm 1.11.0 "LDC - the LLVM D compiler (1.11.0):")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Build targets in project: 1
Found ninja-1.8.2 at /usr/local/bin/ninja

|> ninja
[0/1] Regenerating build files.
The Meson build system
Version: 0.48.0.dev1
Source dir: /home/users/russel/Repositories/Git/Forks/TinyEndian
Build dir: /Users/russel/BuildArea/TinyEndian
Build type: native build
Project name: tinyendian
Project version: 0.2.0
Native D compiler: ldc2 (llvm 1.11.0 "LDC - the LLVM D compiler (1.11.0):")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Build targets in project: 1
Found ninja-1.8.2 at /usr/local/bin/ninja
[2/2] Linking target libtinyendian.0.dylib.

|> ninja install
[0/1] Installing files.
Installing libtinyendian.0.dylib to /Users/russel/Built/lib
Installing /home/users/russel/Repositories/Git/Forks/TinyEndian/source/tinyendian.d to /Users/russel/Built/include/d
Installing /Users/russel/BuildArea/TinyEndian/meson-private/tinyendian.pc to /Users/russel/Built/lib/pkgconfig
```

File: meson.build [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
-)
+# -*- mode: python; -*-
 
-project_soversion    = '0'
+project(
+    'tinyendian',
+    'd',
+    meson_version: '>=0.41',
+    version: '0.2.0',
+    default_options: ['buildtype=release'],
```
Comment (ximion) [inline meson.build]: I am not sure if changing the default is wise here - it's quite unexpected to have one project build differently, especially in a mode that doesn't include debugging symbols.

File: meson.build [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
-        [tinyendian_src],
-        include_directories: [src_dir],
-        install: true,
-        pic: true
+tinyendian_lib = library(
+    meson.project_name(),
+    [tinyendian_src],
+    include_directories: [src_dir],
+    version: meson.project_version(),
+    pic: true,
```
Comment (ximion) [inline meson.build]: `pic: true` is superfluous and could be dropped.

**Comment by BBasile** [comment]
Rebase is required anyway due to a reg in Meson.

**Comment by BBasile** [comment]
Should this be merged ?

**Comment by russel** [comment]
I'd say yes. Obviously. :-) The build seems to work fine with Meson 0.49.0 and Ninja 1.8.2.

**Review by ghost — APPROVED** [review]

**Pull request merged**